sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -115,9 +113,26 @@ class SQLContext private[sql](
   }
 
   /**
-   * @return Spark SQL configuration
+   * Per-session state, e.g. configuration, functions, temporary tables etc.
    */
-  protected[sql] lazy val conf = new SQLConf
+  @transient
+  protected[sql] val sessionState: SessionState = new SessionState(self)
+  protected[sql] def conf: SQLConf = sessionState.conf
```

Comment:
we will probably need to remove these once we create SparkSession and pass that around - but for now it is ok to leave them here.